Error: Cannot find module 'wake-lock'

`wakelock` dependency was made optional in #160 to fix non dependency installing issues on non macOS systems. But we still statically import the module, which throws an error at build time: `Error: Cannot find module 'wake-lock'`.

I see two options to fix this error:

1. We can use [dynamic imports](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import) and import the module inside the macOS specific condition block:

https://github.com/alexrudd2/saxi/blob/bdf77132a5ac2a61c54f6ead8a22d953a50b3296/src/server.ts#L99

2. We can remove wake-lock entirely as it makes little sense to have a single-platform solution and there are good alternatives out there like Caffeine or Amphetamine (which we could reference in the README).

--- a/.github/workflows/lint.yml
+++ b/.github/workflows/lint.yml
@@ -1,14 +1,11 @@
 # This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
 # For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
-
 name: Node.js CI
-
 on:
   push:
     branches: [ "main" ]
   pull_request:
     branches: [ "main" ]
-
 jobs:
   build:
     name: ${{ matrix.os }} - ${{ matrix.node-version }}
@@ -16,7 +13,12 @@ jobs:
       fail-fast: false
       matrix:
         node-version: [20.x, 22.x, 23.x]
-        os: ["windows-latest", "ubuntu-latest"]
+        os: ["windows-latest", "ubuntu-latest", "macos-latest"]
+        exclude:
+          - node-version: 20.x
+            os: macos-latest
+          - node-version: 22.x
+            os: macos-latest
     runs-on: ${{ matrix.os }}
     steps:
     - uses: actions/checkout@v4
@@ -27,7 +29,5 @@ jobs:
         cache: 'npm'
     - run: npm ci
     - run: npm test
-
 permissions:
   contents: read
-  pull-requests: read

--- a/src/server.ts
+++ b/src/server.ts
@@ -13,7 +13,6 @@ import type { PortInfo } from "@serialport/bindings-interface";
 import cors from "cors";
 import type { Request, Response } from "express";
 import express from "express";
-import { WakeLock } from "wake-lock";
 import type WebSocket from 'ws';
 import { WebSocketServer } from 'ws';
 import { EBB, type Hardware } from './ebb.js';
@@ -123,15 +122,19 @@ export async function startServer(port: number, hardware: Hardware = 'v3', com: 
       res.status(200).end();
 
       const begin = Date.now();
-      // biome-ignore lint/suspicious/noExplicitAny: need a new strategy for wakeLock
-      let wakeLock: any;
+      let wakeLock: { release(): void } | null = null;
+      
       // The wake-lock module is macOS-only.
       if (process.platform === 'darwin') {
         try {
+          // Dynamically import wake-lock only on macOS
+          const { WakeLock } = await import("wake-lock");
           wakeLock = new WakeLock("saxi plotting");
-        } catch {
+        } catch (error) {
           console.warn("Couldn't acquire wake lock. Ensure your machine does not sleep during plotting");
         }
+      } else {
+        console.log("Wake lock not available on this platform. Ensure your machine does not sleep during plotting");
       }
 
       try {

